scripts/dev/upstream_merge/upstream_merge.py

```diff
@@ -0,0 +1,274 @@
+            return (0,None)
+        else:
+            return (0,diff_output[1])
+    else:
+        return (1,merge_result[1])
+
+def push_and_PR(git_obj,merge_branch_name,work_item_id):
+    if git_obj.add_remote(git_obj.fork_name, git_obj.fork_url)[0] != 0:
+        print(f"\n    Error adding remote repository {git_obj.fork_name} using {git_obj.fork_url}. Exiting")
+        return (1,f"\n    Error adding remote repository {git_obj.fork_name} using {git_obj.fork_url}. Exiting")
+    
+    if git_obj.push(merge_branch_name, git_obj.fork_name, delete=True)[0] != 0:
+        print(f"\n    Failed to delete branch {merge_branch_name} on {git_obj.fork_name}")
+        return (1, f"\n    Failed to delete branch {merge_branch_name} on {git_obj.fork_name}")
+    
+    if git_obj.push(merge_branch_name, git_obj.fork_name)[0] != 0:
+        print(f"\n    Failed to push branch {merge_branch_name} to {git_obj.fork_name}")
+        return (1, f"\n    Failed to push branch {merge_branch_name} to {git_obj.fork_name}")
+    
+    #if git_obj.create_pull_request("Automated Merge PR",
+#       f"""Merge latest from upstream. No conflicts.
+ 
+    # #AB{work_item_id}
+    
+    # - [ ] bitbake packagefeed-ni-core
+    # - [ ] bitbake packagegroup-ni-desirable
+    # - [ ] bitbake package-index && bitbake nilrt-base-system-image
+    # - [ ] Reimaged a cRIO with the new base image and successfully booted it""",git_obj.local_base_branch,f"Shreejit-03:{merge_branch_name}")[0] != 0:
+    #     print("\n    Error creating the pull request.")
+    #     return (1,"\n    Error creating the pull request.")
+
+    return (0,None)
+    
+def write_email_addresses(email_log_file_name, email_from, email_to):
+    with open(email_log_file_name, "w") as log:
+        log.write(f"From: {email_from}\n")
+        log.write(f"To: {email_to}\n")
+        log.write("Subject: Merge Details\n\n")
+
+def format_status(status, message):
+        if status == 1:
+            return " ... ERRORS", f" ... ERRORS\n    {message or ''}\n", f" ... ERRORS\n    {message or ''}\n\n\n"
+        elif message is None:
+            return " ... OK (no changes)", "", " ... OK (no changes)\n\n\n"
+        else:
+            return " ... OK", "", f" ... OK\n    {message}\n\n\n"
+        
+def format_merge_report(merge_report, email_log_level):
+    min_detail = ""
+    error_detail = ""
+    additional_detail = ""
+    
+    build_and_test_detail = merge_report.pop("Build and Test")
+
+    for git_obj, (status, message) in merge_report.items():
+        min_line, error_line, additional_line = format_status(status, message)
+        min_detail += f"{git_obj.local_repo}\n{min_line}\n"
+        if error_line != "":
+            error_detail += f"{git_obj.local_repo}\n{error_line}\n"
+        additional_detail += f"{git_obj.local_repo}\n\n{additional_line}"
+        if build_and_test_detail[0] == 0 and status == 0 and message is not None:
+            git_obj_push_details = push_and_PR_details[git_obj]
```

Comment:
Can you explain this line? What is `push_and_PR_details[git_obj]`? Is `push_and_PR_details` a `dict`, and if so, where is it even being created?